presto-main/src/test/java/com/facebook/presto/memory/TestMemoryPools.java

```diff
@@ -105,4 +175,76 @@ public static boolean isWaitingForMemory(List<Driver> drivers)
         }
         return false;
     }
+
+    private class RevocableMemoryOperator
+            implements Operator
```

Comment:
I would introduce an `AbstractOperator` which would have default implementation for most of these methods